src/databricks/labs/ucx/source_code/base.py

```diff
@@ -127,7 +127,7 @@ class Convention(Advice):
 
 class Linter:
     @abstractmethod
-    def lint(self, code: str) -> Iterable[Advice]: ...
+    def lint(self, code: str, session_state: CurrentSessionState) -> Iterable[Advice]: ...
```

Comment:
why do we need to change the interface? why it can't be injected via constructor only to those linters than need to be aware of the current session state? revert.